Source/Filter/BilinearFilter.cpp

```diff
@@ -46,8 +46,9 @@ namespace {
         }
         // we use old_s here to make sure the stencil
         // is corrent even when npass = 0
-        stencil = old_s;
-        stencil[0] *= 0.5; // because we will use it twice
+        old_s[0] *= 0.5; // because we will use it twice
+        stencil.resize(old_s.size());
+        Gpu::copyAsync(Gpu::hostToDevice,old_s.begin(),old_s.end(),stencil.begin());
```

Comment:
I fear that `old_s` will run out of scope and we could be touching invalid source memory here during the copy.

Reply:
Yes, I believe you are right.